Base/Logic/vtkSlicerApplicationLogic.cxx

```diff
@@ -163,16 +152,20 @@ void vtkSlicerApplicationLogic::PrintSelf(ostream& os, vtkIndent indent)
 //----------------------------------------------------------------------------
 void vtkSlicerApplicationLogic::CreateProcessingThread()
 {
-  if (this->ProcessingThreadId == -1)
+  if (this->ProcessingThreadIDs.size() == 0)
     {
     this->ProcessingThreadActiveLock.lock();
     this->ProcessingThreadActive = true;
     this->ProcessingThreadActiveLock.unlock();
 
-    this->ProcessingThreadId
-      = this->ProcessingThreader
-      ->SpawnThread(vtkSlicerApplicationLogic::ProcessingThreaderCallback,
-                    this);
+    // Start processing threads (TODO: make the number of threads a setting)
+    int numberOfProcessingThreads = 2;
```

Comment:
Question:
* What should be the default ? Would it break existing workflow to have this than 1 by default ?
* An entry in the Module settings panel could be added ? 
* Should we also support setting the number of worker thread through command line and/or env variable ?

Reply:
I just came to know about this PR. Great initiative. I'm glad to see efforts to make the way for parallel processing.

IMO, the safest would be to leave the default to 1 and add both an entry in the module settings and a command line flag to change the behavior. As a downside, this approach may not expose the feature to the users enough to make a good detection of benefits and problems.

Reply:
We could use 4 as a default value. That would make users feel like everything runs in parallel and would probably not overwhelm most configurations. It would need to be able to change this in application settings.

Reply:
I think an "auto" setting by default, that sets the number to the number of CPUs minus one could be good.

Note though that having a single CLI process is not always a bad thing, since most ITK algorithms that have been the traditional key CLI use case are already multithreaded and running multiples could lead to thrashing.

Perhaps another extension could be to add a CLI flag in the XML that declares if they are multithreaded and make it so that only one of those is run at a time.  This would be easy to add to the built-in CLIs.